Document duplicate project id detection (#12186)
gradle/gradle: subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc

The practical effect is that now all <<java_platform_plugin.adoc#sec:java_platform_consumption,platform dependencies>> have to be declared as such.
Before, platform dependencies also worked, accidentally, when the `platform()` keyword was omitted for local platforms or platforms published with Gradle Module Metadata.

==== Duplicate project names may cause publication to fail

> A fix done to disambiguate projects https://github.com/gradle/gradle/issues/847[having the same name] in dependency resolution made it possible to start publishing projects with identical coordinates.

Reviewer: ```suggestion
A fix done to disambiguate projects https://github.com/gradle/gradle/issues/847[with the same name] allows publishing projects with identical coordinates.
```

This could lead to invalid publications as multiple modules would overwrite each other.

Given the above, a duplicate identifier check was added to prevent invalid publications.

> In some corner cases, the check can prove too eager and may need some action from the build author.

Reviewer: ```suggestion
In cases where published subprojects are involved in such a conflict, an action from the build author is required.
```

----
Project :server:common has the same (groupId, artifactId) as :client:common.
You should set both the groupId and artifactId of the publication or opt out by adding the org.gradle.dependency.duplicate.project.detection system property to 'false'.
----

> If your build fails with a message as above, you have two options:

Reviewer: ```suggestion
If your build fails with a message like the above, you should resolve the conflict either by changing the corresponding project path or by adjusting the `groupId` and `artifactId` in the publication to resolve ambiguities.

As a temporary fix, you can also disable the new check by setting `-Dorg.gradle.dependency.duplicate.project.detection=false` on the CLI or by adding `systemProp.org.gradle.dependency.duplicate.project.detection=false` to `gradle.properties`. This will be removed in future versions.
```

Remove the below 👇
